src/api/fetchContent.ts

```diff
@@ -54,17 +54,27 @@ export function fetchContent<I extends VersionedSlotId, C extends JsonObject>(
                 : {...fetchOptions, version: version},
         );
 
-    if (fallback !== undefined) {
-        return promise.catch(
-            error => {
-                if (logger !== undefined) {
-                    logger.error(`Failed to fetch content for slot "${id}@${version}": ${formatCause(error)}`);
-                }
+    return promise.catch(
+        async error => {
+            if (logger !== undefined) {
+                logger.error(`Failed to fetch content for slot "${id}@${version}": ${formatCause(error)}`);
+            }
 
+            if (fallback !== undefined) {
                 return {content: fallback};
-            },
-        );
-    }
+            }
+
+            const locale = 'preferredLocale' in fetchOptions
+                ? fetchOptions.preferredLocale
+                : null;
+
+            const file = `${locale === null ? '' : `${locale}/`}${slotId}.json`;
 
-    return promise;
+            try {
+                return {content: (await import(`@croct/content/slot/${file}`)).default};
```

Comment:
This might cause problems when using the plug with a bundler since they won't be able to resolve the dynamic import.

Vite, for example, will straight out refuse to bundle this as is and require a plugin to hook into the process and ignore the errors. The value from the fallback won't be loadable from the bundle even with the plugin. It would need to be ``import(/* @vite-ignore */ `@croct/content/slot/${file}`)``.

Esbuild will have a similar problem. And maintaining comments to have every bundler ignore this will:
- Be a pain to maintain;
- Only make the code fail at runtime instead of during build since the bundle won't have the code for the module